pty/ptytest/ptytest.go

```diff
@@ -72,36 +54,60 @@ func create(t *testing.T, ptty pty.PTY, name string) *PTY {
 	go func() {
 		defer close(copyDone)
 		_, err := io.Copy(w, ptty.Output())
-		_ = out.closeErr(err)
+		logf(t, name, "copy done: %v", err)
+		logf(t, name, "closing out")
+		err = out.closeErr(err)
+		logf(t, name, "closed out: %v", err)
 	}()
+
+	tpty := &PTY{
+		t:    t,
+		PTY:  ptty,
+		out:  out,
+		name: name,
+
+		runeReader: bufio.NewReaderSize(out, utf8.UTFMax),
+	}
+	// Ensure pty is cleaned up at the end of test.
 	t.Cleanup(func() {
-		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
+		_ = tpty.Close()
+	})
+
+	logClose := func(name string, c io.Closer) {
+		tpty.logf("closing %s", name)
+		err := c.Close()
+		tpty.logf("closed %s: %v", name, err)
+	}
+	// Set the actual close function for the tpty.
+	tpty.close = func(reason string) error {
+		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitMedium)
```

Comment:
nit: in theory, this can be also `WaitShort`.

Reply:
I was paranoid since we're waiting on two goroutines to close, but we can always bump it if it becomes a problem. I'll go with short.